src/EditorFeatures/Core/Implementation/EditAndContinue/IEditAndContinueSolutionProvider.cs

```diff
@@ -0,0 +1,18 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+namespace Microsoft.CodeAnalysis.Editor.Implementation.EditAndContinue
+{
+    /// <summary>
+    /// Retrieves the <see cref="Solution"/> snapshot that corresponds to the current state of the debuggee.
+    /// This snapshot contains changes successfully applied during EnC/Hot Reload.
+    /// </summary>
+    /// <remarks>
+    /// This is temporarily available to in-proc XAML External Access APIs and should be revisited once XAML moves to OOP LSP.
+    /// </remarks>
+    internal interface IEditAndContinueSolutionProvider
+    {
+        Solution? CommittedSolution { get; }
```

Comment:
>Solution? CommittedSolution { get; } [](http://example.com/codeflow?start=8&length=36)

I think I will need an event for when CommittedSolution changes. Any other event I use for when hot reload happens may not fire at the right time.